File: benchmark_phase_c/bench_nested.cs
using System;

class Program
{
    static int MakeDeep(int aParam)
    {
        int a = aParam;
        Func<int, int> mid = (bParam) =>
        {
            int b = bParam;
            Func<int, int> inner = (c) => a + b + c;
            return inner(10);
        };
        return mid(1);
    }

    static int BenchNested(int n)
    {
        int total = 0;
        int i = 0;
        while (i < n)
        {
            total = total + (MakeDeep(i) % 1000);
            i = i + 1;
        }
        return total;
    }

    static void Main(string[] args)
    {
        int n = int.Parse(args[0]);
        Console.WriteLine(BenchNested(n));
    }
}

File: benchmark_phase_c/bench_nested.py
import sys


def make_deep(a_param):
    a = a_param

    def mid(b_param):
        b = b_param

        def inner(c):
            return a + b + c

        return inner(10)

    return mid(1)


def bench_nested(n):
    total = 0
    i = 0
    while i < n:
        total = total + (make_deep(i) % 1000)
        i = i + 1
    return total


if __name__ == '__main__':
    n = int(sys.argv[1])
    print(bench_nested(n))
